kivy/core/window/__init__.py

```diff
@@ -404,6 +404,14 @@ def _set_size(self, size):
         else:
             return False
 
+    def _recalc_dpi(self):
+        # calculate density
+        w, h = self.system_size
+        sw, sh = self._win._get_gl_size()
```

Comment:
This function is specific to window_sdl2, won't work in anything else as that does not exist else where. 
This is why this function was only in window_sdl2

Reply:
This code runs on various platforms too so there needs to be a check `platform == osx` too.

Reply:
I don't think this would be specific to osx. Windows (and probably ubuntu) also support screens with different `dpi`. E.g. I have a 4k screen on my laptop, but my external monitor is a normal HD screen. 

Although I'm not sure sdl2 currently listens to these dpi changes on windows. See https://discourse.libsdl.org/t/wip-windows-high-dpi-support/22446 (`SDL_WINDOW_ALLOW_HIGHDPI`).

Reply:
@mahomahomaho search for def _get_gl_size it is only defined here https://github.com/kivy/kivy/blob/master/kivy/core/window/_window_sdl2.pyx#L291

@matham I forgot that you could manually set DPI in windows, so you are right.

In light of this probably a better idea is to implement `get_content_size` abstract function in window.__init__ and have it be overridden in specific implementations like window_sdl2 etc.

Even if SDL2 does not directly support DPI on windows yet, we could always call direct WINAPI's and OSX API with `if platform ==` checks. We do not have to depend on sdl2 to implement this considering we also have other window providers too.

Reply:
search for def _get_gl_size it is only defined here https://github.com/kivy/kivy/blob/master/kivy/core/window/_window_sdl2.pyx#L291

Reply:
Hmm, if _get_gl_size is defined only in SDL2 provider, but is used outside it (in _get_size), then it's strange. 

Does it mean that other providers are not used at all, so nobody reported error about undefined _get_gl_size?

Reply:
>Does it mean that other providers are not used at all, so nobody reported error about undefined _get_gl_size

Probably. I don't think anyone uses pygame, and in fact we plan to remove it after this release.

Reply:
The Rpi and the X11 window provider on linux are the ones that I can think of people using... maybe people switch to sdl2 just cause these other providers were broken?

Anyways,  we had identified a problem, makes sense to get that fixed rather than keep ignoring.

Reply:
So maybe just move this implementation to sdl2_window.py and put a placeholder here so we can merge?